### PR TITLE
54360 - Add Set command

### PR DESCRIPTION
**Pull request by volllly**

### CODE REVIEW

File: src/fiskaltrust.Launcher/Commands/ConfigCommand.cs [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+            {
+                Log.Warning("Launcher configuration file {file} does not exist. Creating new file.", LauncherConfigurationFile);
+                launcherConfiguration = new LauncherConfiguration();
+            }
+            else
+            {
+                try
+                {
+                    launcherConfiguration = LauncherConfiguration.Deserialize(await File.ReadAllTextAsync(LauncherConfigurationFile));
+                    try
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Commands/ConfigCommand.cs:66]: We could remove at least one of these try-catch blocks, right? 😅

File: src/fiskaltrust.Launcher/Commands/ConfigCommand.cs [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+                    {
+                        try
+                        {
+                            launcherConfiguration.Decrypt(CipyerAccessToken);
+                        }
+                        catch (Exception e)
+                        {
+                            Log.Warning(e, "Error decrypring launcher configuration file.");
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Commands/ConfigCommand.cs:74]: Typos (also below)

File: src/fiskaltrust.Launcher/Commands/ConfigCommand.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        }
+    }
+
+    public class ConfigSetCommandHandler : ICommandHandler
+    {
+        public LauncherConfiguration ArgsLauncherConfiguration { get; set; } = null!;
+        public string LauncherConfigurationFile { get; set; } = null!;
+
+        public Guid? CipyerCashboxId { get; set; }
+        public string? CipyerAccessToken { get; set; }
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Commands/ConfigCommand.cs:45]: What does "Cipyer" mean?

File: src/fiskaltrust.Launcher/Commands/ConfigCommand.cs [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+                    }
+                    catch (Exception e)
+                    {
+                        Log.Warning(e, "Error decrypring launcher configuration file.");
+                    }
+                    remoteConfiguration?.DisableDefaults();
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, "Could not read remote launcher configuration.");
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Commands/ConfigCommand.cs:236]: I think we're using this logic (decrypting the config) 3x in this file, might make sense to put it into a method

File: src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs [hunk 5]
```diff
@@ -11,12 +12,9 @@ public static class CashBoxConfigurationExt
         private const string ENCRYPTION_SUFFIX = "_encrypted";
         private static readonly List<string> _configKeyToEncrypt = new() { "connectionstring" };
 
-        public static void Decrypt(this ftCashBoxConfiguration cashboxConfiguration, ECDiffieHellman clientEcdh, LauncherConfiguration launcherConfiguration)
+        public static void Decrypt(this ftCashBoxConfiguration cashboxConfiguration, LauncherConfiguration launcherConfiguration)
```
Comment (TSchmiedlechner) [inline src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs:15]: I'm not sure why we changed this now 😅 the cashbox config is coming from the configuration service, which uses the shared secret (and we should absolutely stick to that, IMO). It's not related to locally encrypting the Launcher config, right? 

Can we move this back to the shared secret logic?

**Review by TSchmiedlechner — CHANGES_REQUESTED** [review]

File: src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs (hunk 5 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs:15]: @volllly or what were the reasons for changing this? s😅

File: src/fiskaltrust.Launcher/Commands/ConfigCommand.cs (hunk 3 above)
Comment (volllly) [reply src/fiskaltrust.Launcher/Commands/ConfigCommand.cs:45]: Should be Cipher 😅 as in used for encryption/decryption...
I was searching for a word that means both encryption and decryption at the same time...

File: src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs (hunk 5 above)
Comment (volllly) [reply src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs:15]: The shared secret is now computed once in the encryption helper directly and stored in a static... should we pull that out for better clarity?

Comment (volllly) [reply src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs]: @TSchmiedlechner I pulled the secret out of the Helper 😁 I think its clearer now

Comment (TSchmiedlechner) [reply src/fiskaltrust.Launcher/Configuration/CashBoxConfigurationExt.cs]: Thanks 😁 I was confused by the fact that we didn't store the key, but you're completely right, this was also missing before. 
We'll add this in a separate PR, I guess :)

**Pull request merged**